### PR TITLE
[pr-deploy] Handle all dist results

### PR DESCRIPTION
**Pull request by estherkim**
- Handles if the Travis dist upload is skipped or errors
- Fixes bug where files were being written to `gs://amp-test-websites-1/dist/` instead of `gs://amp-test-websites-1/<travisbuild>/dist/`

### CODE REVIEW

File: pr-deploy/src/app.ts [hunk 1]
```diff
@@ -45,15 +45,23 @@ function initializeCheck(app: Application) {
...
-      const {travisBuild, headSha, exitCode} = request.params;
+      const {travisBuild, headSha, result} = request.params;
       const github = await app.auth(Number(process.env.INSTALLATION_ID));
       const pr = new PullRequest(github, headSha);
-
-      exitCode == 0
-        ? await pr.enableDeploymentCheck(travisBuild)
-        : await pr.errorCompilationCheck();
+      switch (result) {
+        case ('success'):
```
Comment (danielrozenberg) [inline pr-deploy/src/app.ts]: Are these parenthesis required? :-o

File: pr-deploy/src/app.ts [hunk 2]
```diff
@@ -64,20 +72,22 @@ function initializeRouter(app: Application) {
  */
 function initializeDeployment(app: Application) {
   app.on('check_run.requested_action', async context => {
+        if (context.payload.check_run.name != process.env.GH_CHECK){
```
Comment (danielrozenberg) [inline pr-deploy/src/app.ts]: Nit: spacing (before and in between `){`)

File: pr-deploy/src/github.ts [hunk 3]
```diff
@@ -149,11 +175,12 @@ export class PullRequest {
       status: 'completed',
       conclusion: 'neutral',
       output: {
-        title: 'Deployment error.',
-        summary: 'Sorry, there was an error creating a test site.',
-        text: error.message,
+        title: 'Build skipped.',
+        summary: 'Sorry, a test site cannot be created because the ' +
```
Comment (danielrozenberg) [inline pr-deploy/src/github.ts]: I don't think you need to apologize here ;)

File: pr-deploy/src/github.ts [hunk 4]
```diff
@@ -149,11 +175,12 @@ export class PullRequest {
...
+         'compilation step was skipped in Travis. This happens if ' +
```
Comment (danielrozenberg) [inline pr-deploy/src/github.ts]: nit: %s:/if/when/

File: pr-deploy/src/github.ts [hunk 5]
```diff
@@ -149,11 +175,12 @@ export class PullRequest {
...
+         'a PR only includes documentation changes. Please check the Travis ' +
```
Comment (danielrozenberg) [inline pr-deploy/src/github.ts]: :%s/documentation changes/non-code changes, such as documentation/

File: pr-deploy/src/github.ts [hunk 6]
```diff
@@ -136,10 +161,11 @@ export class PullRequest {
 
     return this.github.checks.update(params);
   }
+
   /**
-   * Fail the check if any part of the deployment fails
+   * Set check to 'neutral' if dist is skipped
```
Comment (danielrozenberg) [inline pr-deploy/src/github.ts]: Nit: period at end

File: pr-deploy/src/app.ts (hunk 1 above)
Comment (estherkim) [reply pr-deploy/src/app.ts]: ooopsies

**Pull request merged**